### PR TITLE
Treatment for many and only parameters

### PR DESCRIPTION
**Pull request by 4lissonsilveira**
- A small treatment to avoid conflict with only and exclude params
has been created.
- An If has been created to verify if many parameter is boolean.

Related: #406

### CODE REVIEW

**Comment by 4lissonsilveira** [comment]
@sloria do you think that is a good approach? Could you(or another contributor) give me any suggestions for improvements?

Thanks!

File: marshmallow/fields.py [hunk 1]
```diff
@@ -386,6 +386,22 @@ def __init__(self, nested, default=missing_, exclude=tuple(), only=None, **kwarg
         self.many = kwargs.get('many', False)
         self.__schema = None  # Cached Schema instance
         self.__updated_fields = False
+
+        if not isinstance(self.many, bool):
+            raise ValueError('Many parameter must be boolean')
```
Comment (lafrech) [inline marshmallow/fields.py:391]: `many` does not have to be a boolean. You can pass anything, as long as it evaluates to a either truthy value if you expect a collection or a falsy value if you expect a single element.

File: marshmallow/fields.py [hunk 2]
```diff
@@ -386,6 +386,22 @@ def __init__(self, nested, default=missing_, exclude=tuple(), only=None, **kwarg
...
+
+        if only:
+            if isinstance(only, basestring):
+                only = [only]
+
+            # Catch fields used at the same time
+            # into only and exclude parameters
+            fields = [item for item in only if item in list(exclude)]
```
Comment (lafrech) [inline marshmallow/fields.py:399]: You could use set intersections to do that.

**Comment by lafrech** [comment]
I'm not sure we're willing to add this kind of validation.

- You may want to pass anything as `many`. It does not have to be a boolean. It will evaluate to `True` or `False`.

- I believe `only` has priority over `exclude` (or conversely, I don't remember). Raising an exception when a field is in both lists may break some workflows. However, I agree that relying on this may be a bit scary, and in some cases, it could hide an error. And the fact that I can't remember which one takes priority and I can't find it quickly in the docs makes me think it could be sensible to prevent it. I suppose the assumption is "Since it can be solved by an explicit priority and it creates no issue, why prevent it? The user should be aware or what he is doing." Same applies to `many`.

**Comment by deckar01** [comment]
These seem like good warnings for new users, but would likely be burdensome errors for existing users.

**Comment by sloria** [comment]
Closing this in favor of #826

**Pull request closed without merging**